tests/unittests/caffe2ImporterTest.cpp

```diff
@@ -61,6 +61,131 @@ TEST(caffe2, importConv) {
+    updateInputsByName(ctx, &mod, {"inputs"}, {&inputs});
+  }
+
+  // High level check on the content of the graph. We have 1 conv and 1 save.
+  EXPECT_EQ(F->getNodes().size(), 2);
+  auto *saveNode = getSaveNodeFromDest(output);
+  auto *convNode =
+      llvm::dyn_cast<ConvolutionNode>(saveNode->getInput().getNode());
+  ASSERT_TRUE(convNode);
+
+  // We have 2 placeholders:  1 input and 1 output.
+  EXPECT_EQ(mod.getPlaceholders().size(), 2);
+  // We have 2 constants: Weights and bias.
+  EXPECT_EQ(mod.getConstants().size(), 2);
+}
+
+/// Test loading MaxPool with NHWC order input.
+TEST(caffe2, maxPoolNHWC) {
+  ExecutionEngine EE{BackendKind::Interpreter};
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  std::string NetDescFilename(
+      "tests/models/caffe2Models/maxpool_nhwc_predict_net.pbtxt");
+  std::string NetWeightFilename(
+      "tests/models/caffe2Models/empty_init_net.pbtxt");
+
+  Placeholder *output;
+  Context ctx;
+
+  Tensor inputs(ElemKind::FloatTy, {1, 3, 3, 1});
+  inputs.getHandle().randomize(-3.0, 3.0, mod.getPRNG());
+
+  // Destroy the loader after the graph is loaded since the following execution
+  // will not depend on anyting from the loader.
+  {
+    Caffe2ModelLoader caffe2LD(NetDescFilename, NetWeightFilename, {"inputs"},
+                               {&inputs.getType()}, *F);
+    output = caffe2LD.getSingleOutput();
+
+    ctx.allocate(mod.getPlaceholders());
+    updateInputsByName(ctx, &mod, {"inputs"}, {&inputs});
+  }
+
+  // High level check on the content of the graph. We have 1 maxpool and 1 save.
+  EXPECT_EQ(F->getNodes().size(), 2);
+  auto *saveNode = getSaveNodeFromDest(output);
+  auto *maxPoolNode =
+      llvm::dyn_cast<MaxPoolNode>(saveNode->getInput().getNode());
+  ASSERT_TRUE(maxPoolNode);
+
+  // We have 2 placeholders:  1 input and 1 output.
+  EXPECT_EQ(mod.getPlaceholders().size(), 2);
+  // We have 0 constants.
+  EXPECT_EQ(mod.getConstants().size(), 0);
+}
+
+/// Test loading AvgPool with NHWC order input.
+TEST(caffe2, avgPoolNHWC) {
+  ExecutionEngine EE{BackendKind::Interpreter};
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  std::string NetDescFilename(
+      "tests/models/caffe2Models/avgpool_nhwc_predict_net.pbtxt");
+  std::string NetWeightFilename(
+      "tests/models/caffe2Models/empty_init_net.pbtxt");
+
+  Placeholder *output;
+  Context ctx;
+
+  Tensor inputs(ElemKind::FloatTy, {1, 3, 3, 1});
+  inputs.getHandle().randomize(-3.0, 3.0, mod.getPRNG());
+
+  // Destroy the loader after the graph is loaded since the following execution
+  // will not depend on anyting from the loader.
+  {
+    Caffe2ModelLoader caffe2LD(NetDescFilename, NetWeightFilename, {"inputs"},
+                               {&inputs.getType()}, *F);
+    output = caffe2LD.getSingleOutput();
+
+    ctx.allocate(mod.getPlaceholders());
+    updateInputsByName(ctx, &mod, {"inputs"}, {&inputs});
```

Comment:
Why do we need this part if we are not executing anything?

Reply:
Good point! Will clean up this part.